src/web/core/src/main/java/org/geoserver/web/system/status/ConsoleInfoUtils.java

```diff
@@ -0,0 +1,66 @@
+/* (c) 2022 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.web.system.status;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.lang.management.ManagementFactory;
+import java.lang.management.ThreadInfo;
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public interface ConsoleInfoUtils {
+
+    /**
+     * @param lockedMonitors From ThreadMXBean docs: if true, dump all locked monitors
+     * @param lockedSynchronizers From ThreadMXBean docs: if true, dump all locked ownable
+     *     synchronizers
+     * @return large text with information about JVM threads
+     */
+    default String getThreadsInfo(boolean lockedMonitors, boolean lockedSynchronizers) {
+        return Arrays.stream(
+                        ManagementFactory.getThreadMXBean()
+                                .dumpAllThreads(lockedMonitors, lockedSynchronizers))
+                .map(ThreadInfo::toString)
+                .collect(Collectors.joining("\n"));
+    }
+
+    /**
+     * To retrieve the PID of the Java process is used the method ManagementFactory.getRuntimeMXBean().getName()
+     * The returned name usually is the format [pid]@[machinename] but depends on the implementation of the JVM
+     *
+     * @return the histo memory dump in the format produced by the jmap -histo:live
+     * */
+    default String getHistoMemoryDump() {
+        try {
+            int pid =
+                    Optional.ofNullable(ManagementFactory.getRuntimeMXBean().getName())
+                            .map(processName -> Arrays.asList(processName.split("@")))
+                            .filter(elements -> !elements.isEmpty())
+                            .map(maybePid -> Integer.valueOf(maybePid.get(0)))
+                            .orElseThrow(
+                                    () ->
+                                            new IOException(
+                                                    "Problem on getting the PID for the java process"));
+
+            final Process jmapProcess =
+                    Runtime.getRuntime().exec(String.format("jmap -histo:live %d", pid));
```

Comment:
Oracle lists `jmap` as 
> Note:This command is experimental and unsupported.

on https://docs.oracle.com/en/java/javase/11/tools/jmap.html

So I'm not sure if it's a good idea to rely on it being available and/or accessible to the user geoserver is running from.